src/main/proto/ga4gh/genotype_phenotype.proto

```diff
@@ -0,0 +1,159 @@
+To enable standardized integration, this schema relies heavily on
+OntologyTerms, for typing phenotype, genomic features, and levels
+of evidence.  Suggested ontologies to leverage include (with browser links):
+
+  - Human Phenotype Ontology (HPO): http://www.ontobee.org/browser/index.php?o=hp
+  - Disease Ontology (DO): http://purl.obolibrary.org/obo/DOID_4
+  - Sequence Ontology (SO): http://www.sequenceontology.org/browser/
+  - Evidence Code Ontology (ECO): http://www.ontobee.org/browser/index.php?o=ECO
+  - Phenotypic Qualities (PATO): http://www.ontobee.org/browser/index.php?o=PATO
+*/
+
+// The top level container for phenotype association data.
+message PhenotypeAssociationSet {
+    // The phenotype association set ID.
+    string id = 1;
+
+    // The phenotype association set name.
+    string name = 2;
+
+    // The ID of the dataset this phenotype association set belongs to.
+    string dataset_id = 3;
+
+    // Optional additional information for this phenotype association set.
+    map<string, google.protobuf.ListValue> info = 4;
+}
+
+// The context in which a genotype gives rise to a phenotype.
+// This is fairly open-ended; as a stub we have a simple ontology term.
+// For example, a controlled term for a drug, or perhaps an instance of a
+// complex environment including temperature and air quality, or perhaps
+// the anatomical environment (gut vs tissue type vs whole organism).
+message EnvironmentalContext {
+    // The Environment ID.
+    string id = 1;
+
+    // Examples of some environment types could be drawn from:
+    // Ontology for Biomedical Investigations (OBI): http://purl.obofoundry.org/obo/obi/browse
+    // Chemical Entities of Interest (ChEBI): http://www.ontobee.org/browser/index.php?o=chebi
+    // Environment Ontology (ENVO):  http://www.ontobee.org/browser/index.php?o=ENVO
+    // Anatomy (Uberon): http://www.ontobee.org/browser/index.php?o=uberon
+    OntologyTerm environment_type = 2;
+
+    // A textual description of the environment. This is used to complement
+    // the structured description in the environmentType field
+    string description = 3;
+}
+
+// An association to a phenotype and related information.
+// This record is intended primarily to be used in conjunction with variants, but
+// the record can also be composed with other kinds of entities such as diseases
+message PhenotypeInstance {
+    // The Phenotype ID.
+    string id = 1;
+    
+    // HPO is recommended
+    OntologyTerm type = 2;
+
+    // PATO is recommended.  Often this qualifier might be for abnormal/normal,
+    // or severity.
+    // For example, severe: http://purl.obolibrary.org/obo/PATO_0000396
+    // or abnormal: http://purl.obolibrary.org/obo/PATO_0000460
+    repeated OntologyTerm qualifier = 3;
+
+    // HPO is recommended, for example, subclasses of
+    // http://purl.obolibrary.org/obo/HP_0011007
+    // TODO: also allow quantitative recording?
+    OntologyTerm age_of_onset = 4;
+
```

Comment:
The time specification in `PhenotypeInstance.age_of_onset`  is not well defined:

* elsewhere in the schema time attributes are defined in ISO8601
* the [HPO link](http://purl.obolibrary.org/obo/PATO_00004) does not resolve
* an optimal implementation would allow both the use of standard ISO time (where age would be of the subtype "duration") and ontology terms for age classes ("childhood onset" or such), as is also discussed in the [Phenopackets](https://github.com/phenopackets/phenopacket-format/wiki/Overview-of-Phenotype-Exchange-Format) draft (though also incomplete there).

In conclusion, this has either be switched to string, ISO; or a proper documentation of the use has to be provided.